### ISSUE TITLE
Add purpose field to google_network_security_address_group.

### ISSUE THREAD

**Issue by carloshdezbueno** [hashicorp/terraform-provider-google#18624]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Description

Basically the address groups are ment to be used for cloud armor but to be able to use them for it, it needs to be configured with that purpose. Therefore, if the purpose cant be set to "CLOUD_ARMOR", the address group cant be used for it

### New or Affected Resource(s)

* google_network_security_address_group

### Potential Terraform Configuration

```tf 
resource "google_network_security_address_group" "default" {
  name        = "my-address-groups"
  parent      = "projects/my-project-name"
  location    = "us-central1"
  type        = "IPV4"
  purpose = "CLOUD_ARMOR
  capacity    = "100"
  items       = ["208.80.154.224/32"]
}
```

**Comment by BBBmau** [comment]
API Reference Link: https://cloud.google.com/service-mesh/docs/reference/network-security/rest/v1beta1/organizations.locations.addressGroups

**Comment by jbbqqf** [comment]
Hi! Triaging older Network Security enhancement requests — this one duplicates [#18602](https://github.com/hashicorp/terraform-provider-google/issues/18602) ("Add Purpose in google_network_security_address_group"), opened on 2024-06-28 (four days earlier) with the same scope.

Both issues request the same `purpose` enum field on `google_network_security_address_group`; consolidating discussion on #18602 would help maintainers prioritize.

Would you mind closing this in favor of #18602? I'll cross-reference there.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the schema and merge history were verified manually against the `terraform-provider-google` `main` branch.)
